### PR TITLE
[Reviewer: Matt] Add patches

### PR DESCRIPTION
**Pull request by johadalin**
The master branch has the original code from the Centos 7 repo, and the souce tarball check-in from the Centos SRPM (so that we are using the -noapsl version). I do not think this needs reviewing as I made no manual changes.

This PR contains two commits, the first of which will also likely not need a review:

- Adding the two patch files that we decided we need from our Debian builds, and a gitignore to exclude the BUILD, RPM, and SRPM directories.

- Some changes to the spec file to build correctly
  - Including the patches in the build
  - Hard coding '...clearwater1' into the Release string, so that we can distinguish (and depend on) our net-snmp build
  - Disabling the net-snmp tests when building. As per commit message, one test was failing in function we do not use, and our Debian build fails many more tests. As such I think this is OK, as we are doing better than maintaining parity between build systems. (The failure is introduced at some point in the 'add_trap_send_callbacks' patch. If we want to investigate, we should raise an issue, and test to find which section is introducing the error case)

### CODE REVIEW

File: SOURCES/add_trap_send_callbacks [hunk 1]
```diff
@@ -0,0 +1,555 @@
+Description: Adds callbacks on INFORM calls
+
+ net-snmp (5.7.2~dfsg-clearwater3) trusty; urgency=medium
```
Comment (mirw) [inline SOURCES/add_trap_send_callbacks]: Is it right for this to be "~dfsg" or "trusty"?

File: SOURCES/add_trap_send_callbacks [hunk 2]
```diff
@@ -0,0 +1,555 @@
+Description: Adds callbacks on INFORM calls
```
Comment (mirw) [inline SOURCES/add_trap_send_callbacks]: Presumably this is just the same as the Ubuntu version?  Should we try to commonize them... or should we at least refer to the other copy somewhere?

File: SOURCES/inclusiveness.patch [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+-                 * GETBULK over an overridden variable. The bulk-to-next
+-                 * handler re-uses the same request for multiple varbinds,
+-                 * and once inclusive was set, it was never cleared. So, a
+-                 * hack. Instead of setting it to 1, set it to 2, so bulk-to
+-                 * next can clear it later. As of the time of this hack, all
+-                 * checks of this var are boolean checks (not == 1), so this
+-                 * should be safe. Cross your fingers.
+-                 */
+-                request->inclusive = 2;
+-                /*
```
Comment (mirw) [inline SOURCES/inclusiveness.patch]: Why this change?

File: SPECS/net-snmp.spec [hunk 4]
```diff
@@ -462,7 +466,7 @@ cp -f libtool.orig libtool
 # temporary workaround to make test "extending agent functionality with pass" working
 chmod 755 local/passtest
 
-LD_LIBRARY_PATH=${RPM_BUILD_ROOT}/%{_libdir} make test
+#LD_LIBRARY_PATH=${RPM_BUILD_ROOT}/%{_libdir} make test
```
Comment (mirw) [inline SPECS/net-snmp.spec]: Please can we comment here why we don't want to run these tests.

**Comment by mirw** [comment]
Looks good.  A few comments inline above.

File: SOURCES/inclusiveness.patch (hunk 3 above)
Comment (johadalin) [inline SOURCES/inclusiveness.patch]: To record our conversation:

This diff is correctly the addition of a patch file, which removes the lines above from the source code. 
Git diff is highlighting the patch removal lines, making it look as if the patch file has been altered, when it has actually just been directly copied across from our Debian build.
I have double checked in the rpmbuild/BUILD/ code, and the patch is being correctly applied on building the project.

File: SOURCES/add_trap_send_callbacks (hunk 2 above)
Comment (johadalin) [inline SOURCES/add_trap_send_callbacks]: Added comments here, and in clearwater-net-snmp versions of the patches to track that these files are direct copies
